packages/payments-plugin/src/stripe/stripe.controller.ts

```diff
@@ -0,0 +1,125 @@
+            Logger.error(`Error verifying Stripe webhook signature ${signature}: ${e.message}`);
+            throw new BadRequestException('Error verifying Stripe webhook signature');
+        }
+
+        const paymentIntent = event.data.object as Stripe.PaymentIntent;
+        if (!paymentIntent) {
+            throw new BadRequestException('No payment intent in the webhook payload');
+        }
+
+        const { metadata: { channelToken, orderCode, orderId } = {} } = paymentIntent;
+
+        if (event.type === 'payment_intent.payment_failed') {
+            const message = paymentIntent.last_payment_error && paymentIntent.last_payment_error.message;
+            Logger.warn(`Payment for order ${orderCode} failed: ${message}`, loggerCtx);
+            return;
+        }
+
+        if (event.type !== 'payment_intent.succeeded') {
+            // This should never happen as the webhook is configured to receive
+            // payment_intent.succeeded and payment_intent.payment_failed events only
+            Logger.info(`Received ${event.type} status update for order ${orderCode}`, loggerCtx);
+            return;
+        }
+
+        const ctx = await this.createContext(channelToken);
+
+        const transitionToStateResult = await this.orderService.transitionToState(
+            ctx,
+            orderId,
+            'ArrangingPayment',
+        );
+
+        if (transitionToStateResult instanceof OrderStateTransitionError) {
+            Logger.error(
+                `Error transitioning order ${orderCode} to ArrangingPayment state: ${transitionToStateResult.message}`,
+                loggerCtx,
+            );
+            return;
+        }
+
+        const paymentMethod = await this.getPaymentMethod(ctx);
+
+        const addPaymentToOrderResult = await this.orderService.addPaymentToOrder(ctx, orderId, {
+            method: paymentMethod.code,
+            metadata: {
+                paymentIntentId: paymentIntent.id,
+            },
+        });
+
+        if (!(addPaymentToOrderResult instanceof Order)) {
+            Logger.error(`Error adding payment to order ${orderCode}: ${addPaymentToOrderResult.message}`);
+            return;
+        }
+
+        Logger.info(`Stripe payment intent id ${paymentIntent.id} added to order ${orderCode}`, loggerCtx);
+    }
+
+    private async createContext(channelToken: string): Promise<RequestContext> {
+        const channel = await this.channelService.getChannelFromToken(channelToken);
+
+        return new RequestContext({
+            apiType: 'admin',
+            isAuthorized: true,
+            authorizedAsOwnerOnly: false,
+            channel,
+            languageCode: LanguageCode.en,
+        });
+    }
+
+    private async getPaymentMethod(ctx: RequestContext): Promise<PaymentMethod> {
+        const method = await this.connection.getRepository(ctx, PaymentMethod).findOne({
+            where: {
+                code: stripePaymentMethodHandler.code,
```

Comment:
You're right. The code above worked in my tests because yeah, I named the payment method "Stripe" as well.

To be honest, I pretty much copied and pasted the code from the Braintree plugin but in fact I was thinking of retrieving the payment method whose **_handler code_** equates to "stripe" since that never changes. What if we used the query below instead to pick the right method?

```typescript
const method = (await this.connection.getRepository(ctx, PaymentMethod).find())
  .find(method => method.handler.code === stripePaymentMethodHandler.code);
```

From my understanding, you cannot filter, i.e use `where`, `simple-json` columns such as `method.handler` so we'd have to retrieve all payment methods first and then pull the Stripe one. I tested this approach and it works. I believe it has 2 benefits over taking payment method as a parameter from the Storefront:
1. It offers a simpler API
2. The Storefront doesn't need to know about a piece of information that might change at any time, for instance, if an Admin updates the payment method name and code on the Admin UI. Even if `eligiblePaymentMethods` is called, the Storefront would still need to understand how to select Stripe

I'm fine either way but please let me know what you think.

Reply:
Yes, your idea is better for the reasons given. Let's go with that 👍

Reply:
I haven't updated the Braintree plugin to use the same logic - even though it's a simple change - as it would require testing, which I'm not able to do.

Reply:
That's fine, I'll handle that part.